client/src/includes/sidePanel.js

```diff
@@ -99,6 +102,123 @@ export default function initSidePanel() {
     });
   }
 
+  const resizeGrip = document.querySelector('[data-form-side-resize-grip]');
+  const widthInput = document.querySelector('[data-form-side-width-input]');
+  const blankCanvas = document.createElement('canvas');
+
+  const getSidePanelWidthStyles = () => {
+    const sidePanelStyles = getComputedStyle(sidePanelWrapper);
+    const minWidth = parseFloat(sidePanelStyles.minWidth);
+    const maxWidth = parseFloat(sidePanelStyles.maxWidth);
+    const width = parseFloat(sidePanelStyles.width);
+    const range = maxWidth - minWidth;
+
+    return { minWidth, maxWidth, width, range };
+  };
+
+  const setSidePanelWidth = (width) => {
+    const { minWidth, range, width: currentWidth } = getSidePanelWidthStyles();
+    const newWidth = parseFloat(width)
+      ? Math.max(minWidth, Math.min(width, minWidth + range))
+      : currentWidth;
+
+    sidePanelWrapper.style.width = `${newWidth}px`;
+    widthInput.value = ((newWidth - minWidth) / range) * 100;
+    localStorage.setItem('wagtail:side-panel-width', newWidth);
```

Comment:
This will need to be wrapped in a `try/catch` so the code doesn’t explode when localStorage is unavailable.